### DISCUSSION TITLE
What does the number mean?

### DISCUSSION THREAD

**Discussion by wangsiqidahaoren** [GumTreeDiff/gumtree#349] [Q&A] [answered]
I use 'cluster' to extract action clusters and save them as a json file. 
`New cluster:
===
insert-node
---
funcdef [566,863]
to
file_input [0,2391]
at 8
------------
===
insert-node
---
funcdef [566,863]
to
file_input [0,2391]
at 8
===
insert-node
---
name: read_json_with_comments [570,593]
to
funcdef [566,863]
at 0

New cluster:
Unknown cluster type
------------
===
insert-tree
---
parameters [593,604]
    param [594,603]
        name: json_path [594,603]
to
funcdef [566,863]
at 1
...
...`

I wonder to know what does the number mean in '[]'? and what does the num after 'as' mean?

**Comment by jrfaller** [comment, marked as answer]
Hi!

Between [] is the absolute character index of the node, and after at is the index position among the children of the tree.

Cheers.

**Reply by wangsiqidahaoren** [reply]
Thanks for your reply.
